PolicyAdministrationPoint/src/main/java/it/cnr/iit/ucs/pap/PolicyAdministrationPoint.java

```diff
@@ -81,27 +81,29 @@ public String retrievePolicy( String policyId ) {
      * @return true if everything goes OK, false otherwise
      */
     @Override
-    public boolean addPolicy( String policy ) {
+    public String addPolicy( String policy ) {
         Reject.ifBlank( policy );
         PolicyWrapper policyWrapper;
         try {
             policyWrapper = PolicyWrapper.build( policy );
         } catch( PolicyException e ) {
-            return false;
+            return null;
```

Comment:
I think that from now on we shouldn't return null.
What about throwing a PAPException ? (to be created)